### ISSUE TITLE
Can W041 be reinstated as a new option?

### ISSUE THREAD

**Issue by fivegreenapples** [jshint/jshint#3285]
In v2.9.5 (which I realise was 11 months ago now) there was a change to remove W041 ([#3115](https://github.com/jshint/jshint/pull/3115)). So, after what would normally be considered a very minor version bump (2.9.4 to 2.9.5), jshint is no longer warning of an issue that it did previously. This I consider to be a backwards incompatible change. There doesn't appear to be a way to reinstate this behaviour.

I for one liked the previous behaviour but I can see it would be unnecessarily strict for many. But I am a bit unhappy that it was changed without any way to get it back. However, I'd be happy to look at re adding this as a new option if the maintainers would accept a PR for such a change?

**Comment by jugglinmike** [comment]
As noted in [the original bug report](https://github.com/jshint/jshint/issues/3109), we removed that warning because it was inconsistently-implemented and therefor a potential source of confusion. We don't classify relaxation of linting behavior to be backwards incompatible because consumers do not tely on the presence of warnings--only on their absence.

That said, we further rationalized the removal of the warning because it served the same purpose as warning W116 ("Expected '===' and instead saw '=='.") which users can opt in to received via the `eqeqeq` linting option. Can you try enabling that option in your project?

**Comment by fivegreenapples** [comment]
Thanks for responding.

> because consumers do not tely on the presence of warnings--only on their absence

I sort of understand that logic, but by the same token, you could remove literally every warning issued by jshint and it would be backwards compatible. If you did that I don't think you could argue that consumers _only_ rely on the absence of warnings.

In any case, I'm not really arguing for the changes done last year to be reversed, I'm more interested in whether you'd accept a PR for the addition of an option that allowed me to turn on W041 independently of `eqeqeq`. If you wouldn't that's fair enough. I'll move on.

It's not a big deal but it is a change in behaviour that I picked up because we run jshint on our codebase on two different systems (local dev and CI test plans). They are configured to run the same config file but run independent jshint source. One of those had been upgraded but seemingly only by a very minor version change. (I say it's not a big deal, but it would be a big deal if you'd removed a warning I really care about. This one I only slightly care about.)

In answer to your question - turning on `eqeqeq` produces a large bunch of warnings I'm not interested in. Particularly sometimes I want to take advantage of the loosely typed nature of javascript and compare `"123"` against `123` and have it succeed. W041 was a much more specific check for comparison against falsey which picks up possible code errors in a particular case. e.g. did you really mean compare to `null` or are you checking for "is falsey":  `if (foo === null)` vs `if (!foo)`.

In short W041 doesn't serve the same purpose as W116, it is a subset. W116 is very broad. If I was starting a new project I probably would have W116 turned on but then I'd also probably go the whole hog and use Typescript.

**Comment by angeliski** [comment]
I agree with @fivegreenapples , make sense the **removing warning logic** , but in reallity not work like that. In my case, the warning is useful and in version 2.9.5 it's gone. 
Maybe that behaviour, remove warnings, should be consider to be a backwards incompatible change.

What do you think @jugglinmike ? It's worse to that project if we get that way?

**Comment by angeliski** [comment]
Any news about that case @jugglinmike ?

**Comment by angeliski** [comment]
Any news about that case @jugglinmike ?
